### DISCUSSION TITLE
Host key verification failed - HOOBS

### DISCUSSION THREAD

**Discussion by Myth0zz** [AlexGustafsson/homebridge-wol#154] [Questions]
Hello. I just installed the plugin. It successfully pings my Mac Mini and it shows it active in my Homekit page. But when I press the Homekit switch for it to go to sleep, I get the following in the HOOBS logs:

![IMG_4325D46407E3-1](https://user-images.githubusercontent.com/80846065/112653910-8afbf880-8e57-11eb-9254-6e26cf010eba.jpeg)

------------------
Got state change for "My Mac Mini" (192.168.152.188): 0
26/03/2021, 17:14:17 [My Mac Mini] NetworkDevice shutdown cycle started for "My Mac Mini" (192.168.152.188)
26/03/2021, 17:14:17 [My Mac Mini] NetworkDevice "My Mac Mini" (192.168.152.188) went from status "Online" to "Shutting Down"
26/03/2021, 17:14:17 [My Mac Mini] Stopping pinger
26/03/2021, 17:14:17 [My Mac Mini] Attempting to shut down "My Mac Mini" (192.168.152.188) using "ssh"
26/03/2021, 17:14:17 [My Mac Mini] An error occured while trying to shut down "My Mac Mini" (192.168.152.188): Error: Command failed: ssh xxxxxxx@My-Mac-mini.local pmset sleepnow
Host key verification failed.

    at ChildProcess.exithandler (child_process.js:308:12)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Socket.<anonymous> (internal/child_process.js:443:11)
    at Socket.emit (events.js:314:20)
    at Pipe.<anonymous> (net.js:676:12) {
  killed: false,
  code: 255,
  signal: null,
  cmd: 'ssh xxxxxx@My-Mac-mini.local pmset sleepnow',
  stdout: '',
  stderr: 'Host key verification failed.\r\n'
}
---------------

Can you help? Thanks!

**Comment by AlexGustafsson** [comment]
You're getting this error because the user running homebridge has not marked the target computer's public SSH key as trusted. This is typically done interactively the first time you connect to the machine via a terminal.

The best solution is to one time manually SSH into your computer as the user running homebridge:

```
ssh xxxxx@My-Mac-mini.local
```

The easiest solution is to (insecurely) allow any public key from the target:

```json
{
  "shutdownCommand": "ssh -o StrictHostKeyChecking=no xxxxxxx@My-Mac-mini.local pmset sleepnow"
}
```

**Comment by Myth0zz** [comment]
Thanks for the reply. After adding the above, I get: 

26/03/2021, 19:56:31 [My Mac Mini] Error: Command failed: ssh -o StrictHostKeyChecking=no xxxx@My-Mac-mini.local pmset sleepnow
Permission denied, please try again.
Permission denied, please try again.
Received disconnect from 192.168.152.174 port 22:2: Too many authentication failures
Disconnected from 192.168.152.174 port 22

    at ChildProcess.exithandler (child_process.js:308:12)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Socket.<anonymous> (internal/child_process.js:443:11)
    at Socket.emit (events.js:314:20)
    at Pipe.<anonymous> (net.js:676:12) {
  killed: false,
  code: 255,
  signal: null,
  cmd: 'ssh -o StrictHostKeyChecking=no xxxxxx@My-Mac-mini.local pmset sleepnow',
  stdout: '',
  stderr: 'Permission denied, please try again.\r\n' +
    'Permission denied, please try again.\r\n' +
    'Received disconnect from 192.168.152.174 port 22:2: Too many authentication failures\r\n' +
    'Disconnected from 192.168.152.174 port 22\r\n'
}
26/03/2021, 19:56:31 [My Mac Mini] Waiting for shutdown grace time (5000ms) for "My Mac Mini" (192.168.152.188).


ALSO, the Hoobs is pinging every 5 seconds / 5 times my Mac - continuously. Is this gonna have trouble to my machine, or connection in the future? Is it Ok to leave it as is?

**Reply by AlexGustafsson** [reply]
You're not allowed to SSH into the machine. You have to configure authentication. There are countless tutorials on that topic, so I won't cover it here. You're looking for tutorials on how to setup SSH key authentication.

A ping a second is nothing to worry about. Your Mac likely receives hundreds if not thousands of similar requests every minute. If you're interested to see how it affects your Mac (and any other network traffic) in real time, you can run Wireshark (https://www.varonis.com/blog/how-to-use-wireshark/).

**Comment by Myth0zz** [comment]
Forgive me but I don't get it. I have enabled Remote Login in my Mac. I also have created an SSH key in my Mac. At least that's what those tutorials you mention, say that I would do. However I don't see any changes. Could you please give me a link to a valid tutorial? 

And I am sorry for taking up your precious time from replying regarding your plugin.

**Reply by AlexGustafsson** [reply]
You’ll have to use `ssh-copy-id user@target` on the machine running Homebridge to setup key authentication to connect to the target. This assumes you have previously set up a key on the machine running Homebridge.

**Reply by Myth0zz** [reply]
I created a key in homebridge. 

Your identification has been saved in key.
Your public key has been saved in key.pub.
The key fingerprint is:
SHA256:xxxxxxxxxxxshagdsfajhdfhgsfdhgasdfhgsfdZE8I xxxxxx@My-Mac-mini.lan

Can you tell me what to do next?

I tried: 

ssh-copy-id xxxxxxx@My-Mac-mini.local but it returns:

ERROR: No identities found

**Reply by AlexGustafsson** [reply]
There seems to be a mismatch in the user who created the key and the user trying to use it. The error specifying that there’s no identity tells us that it didn’t find any generated key. If you stored the key in a specific location, you may use `-i /path/to/key` to specify the path manually.

See here for more info: https://www.ssh.com/ssh/copy-id

**Reply by Myth0zz** [reply]
Thank you for your help. I never managed to overcome this authentication error. I make the key, I tell it to copy it but it doesn't find it. 

I uninstalled the plugin. Thanks again for your efforts to help me.

**Reply by AlexGustafsson** [reply]
Sorry it didn't end up working for you. I hope to have more time to write guides for use cases like yours. It's on the roadmap (#148). I'll try to get back here once progress has been made on it. Hopefully it will end up more user-friendly than my comments above.

If you end up resolving it, please feel free to comment here or on the linked issue with your solution so that it may be included in the guide to help others in the future.
